### DISCUSSION TITLE
Project Quotes Plugin

### DISCUSSION THREAD

**Discussion by Apfelwurm** [kimai/kimai#2881] [General]
Hi together,
i think about using kimai2 in the future, but to really have the system that i need, it would require the ability to generate quotes directly from it.

As i searched around, i found a [closed Issue](https://github.com/kevinpapst/kimai2/issues/1937) on this topic, which indicates that it is not planned to implement this feature inside of kimai2 in the near future.

Because i had to write a lot of PHP lately, i thought about starting the plugin development for this feature. I can't say for sure if (and how fast) thats going to happen, but since this could be useful for more people, i want to collect requirements and the ideas for how this could be implemented from a users / teams / administrative perpective (since i have no practical experience with using kimai2 productive).

Also, if someone wants to join if i start writing the plugin, just let me know!

Thanks in advance for any input :)

Alex

**Comment by kevinpapst** [comment]
Hi Alex, I rarely need that feature myself, but I'd be happy to join the discussion and help with the plugin.
So if you start the development (there is a [DemoBundle for the quick start](https://github.com/Keleo/DemoBundle)) and have questions: please let me know!

My first quick ideas about the feature: 
- record multiple line items with the quote contents
- allow template customizations (probably concentrate on Twig/PDF first)
- keep a list of quotes for a customer
- allow to convert a quote into a project
